resources-domain/resources-service/package.json

```diff
@@ -28,7 +29,7 @@
     "test:service": "cross-env POSTGRES_PORT=5433 RDS_USERNAME=hrm RDS_PASSWORD=postgres RESOURCES_PASSWORD=resources-password run-s -c docker:compose:test:up db:create:schema test:service:ci:migrate test:service:ci:run docker:compose:test:down",
     "test:service:ci": "RDS_USERNAME=rdsadmin RDS_PASSWORD=postgres RESOURCES_PASSWORD=resources-password run-s db:create:schema test:service:ci:migrate test:service:ci:run",
     "test:service:ci:migrate": "node ./db-migrate",
-    "test:service:ci:run": "cross-env AWS_REGION=us-east-1 CI=true TWILIO_ACCOUNT_SID=ACxxx TWILIO_AUTH_TOKEN=xxxxxx SSM_ENDPOINT=http://localhost:4566 jest --verbose --maxWorkers=1 --forceExit tests/service",
+    "test:service:ci:run": "cross-env AWS_REGION=us-east-1 CI=true TWILIO_ACCOUNT_SID=ACxxx TWILIO_AUTH_TOKEN=xxxxxx SSM_ENDPOINT=http://mock-ssm/ jest --verbose --maxWorkers=1 --forceExit tests/service/",
```

Comment:
I'm not proposing we shouldn't be using localstack in service tests, but it just seemed a very heavyweight dependency to add just to test if one endpoint looked up an SSM parameter and published something to a queue, so I just wanted to see if something simple that we could run inside the test process would do the job for this case.

Reply:
Can convert the tests to localstack if folks prefer that, I was just thinking we could use simpler, in process mocks for testing straightforward cases and only bring localstack into play for testing more complex interactions across several AWS services

Reply:
I tend to think of service tests as tests that run against a full local running stack whenever possible and depend on mocks for unit tests to keep them as lightweight as possible.

I'm not stuck on the idea, but if a future goal is to test both our terraform provisioned infrastructures and service tests in a single path, it feels like this is a step away from that pattern instead of towards it.

It isn't that important for this PR, but something we should probably align on sooner rather than later.

Reply:
My thoughts are 'service tests', as in testing a single service in isolation, should have flexible lightweight mocks who's behaviour can easily be programmed & verified to suit the needs of individual test cases (the exception being the DB, because the interactions with it are just so damn complex that nothing less than the real thing will do).

It's when we go up to testing how multiple services interact that I agree we need the full running local stack, whether it's full E2E tests or testing a subset of highly interrelated services (integration test, domain level test, whatever the nomenclature) - and it's these sort of tests that I thought we would align testing our terraform configurations alongside our application behaviour, not at the level of service tests

Put another way, my mental model always thought of single service tests as 'big unit tests' rather than 'little integration tests', which guided my approach to mocking etc.

Definitely more to dig into than will be figured out on a PR thread - topic for a future Aselo Tech Discussion probably